src/types/migration/fields.ts

```diff
@@ -0,0 +1,56 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { ContentRelationshipField } from "../value/contentRelationship"
+import type { PrismicDocument } from "../value/document"
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { ImageField } from "../value/image"
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { LinkField } from "../value/link"
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { LinkToMediaField } from "../value/linkToMedia"
+
+import type { MigrationAsset } from "./asset"
+import type { PrismicMigrationDocument } from "./document"
+
+export const MigrationFieldType = {
+	Image: "image",
+	LinkToMedia: "linkToMedia",
+} as const
+
+/**
+ * An alternate version of the {@link ImageField} for use with the migration API.
+ */
+export type ImageMigrationField =
+	| (MigrationAsset & {
+			migrationType: typeof MigrationFieldType.Image
+	  })
+	| undefined
+
+/**
+ * An alternate version of the {@link LinkToMediaField} for use with the
+ * migration API.
+ */
+export type LinkToMediaMigrationField =
+	| (MigrationAsset & {
+			migrationType: typeof MigrationFieldType.LinkToMedia
+	  })
+	| undefined
+
+/**
+ * An alternate version of the {@link ContentRelationshipField} for use with the
+ * migration API.
+ */
+export type ContentRelationshipMigrationField =
+	| PrismicDocument
```

Comment:
No, it's not possible to have a resolved migration document at that point as it might not be created yet.

```ts
// ✅ Correct
migration.createDocument({
  data: {
    category: () => migration.getByUID("category", "event")
  }
})

// ❌ Incorrect: The queried document might not be resolved yet
migration.createDocument({
  data: {
    category: migration.getByUID("category", "event")
  }
})
```

Reply:
Thinking more about it, the following should be working though, so let's allow it:

```ts
const eventCategory = migration.createDocument(...)

migration.createDocument({
  data: {
    category: eventCategory
  }
})
```